contrib/node/src/python/pants/contrib/node/tasks/node_repl.py

```diff
@@ -59,10 +59,11 @@ def launch_repl(self, targets):
         json.dump(package, fp, indent=2)
 
       args = self.get_passthru_args()
-      node_repl = self.node_distribution.node_command(args=args)
+      node_repl = self.node_distribution.node_command(
+        args=args, node_paths=node_paths.all_node_paths if node_paths else None)
 
       with pushd(temp_dir):
-        result, npm_install = self.execute_npm(['install'],
+        result, npm_install = self.execute_npm(['install', '--no-optional'],
```

Comment:
Given that this is in multiple places, should it be on a subsystem? Maybe on the distribution?

Reply:
I would hold off changing node_distribution within this PR.  I would like to remodel the node_distribution and unify npm/yarnpkg code path in a separate PR.  Added a TODO comment.